### PR TITLE
APIv2(gateway_balances, channel_authorize): update errors

### PR DESCRIPTION
**Pull request by PeterChen13579**
## High Level Overview of Change
Fixed/added error messages for Gateway_balance(issue https://github.com/XRPLF/rippled/issues/4290 | https://github.com/XRPLF/rippled/issues/4548 ) and channel_authorize(https://github.com/XRPLF/rippled/issues/4289 )

### Context of Change
Added error messages when user sends an incorrect request. (Check the above Github issue link for more info.) Everything is under API Version 2 as it is a breaking change.

### Type of Change

- [ X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ X] Tests (You added tests for code that already exists, or your new feature included in this PR)

Added own unit test for API Version 2;

### CODE REVIEW

**Comment by PeterChen13579** [comment]
@gregtatcam @shawnxie999 Hi guys, I accidentally messed up my git, so I had to create a new PR. This one is exactly the same changes as my old one([https://github.com/XRPLF/rippled/pull/4577](https://github.com/XRPLF/rippled/pull/4577)); please approve this PR if everything looks good. Thank you

**Review by shawnxie999 — APPROVED** [review]

**Review by gregtatcam — APPROVED** [review]
👍

**Comment by intelliot** [comment]
@PeterChen13579 this PR currently has merge conflicts which you'll need to resolve (preferably with a merge commit).

@gregtatcam do you have an opinion on whether it's OK for this PR to be merged before #4626?

(I have no preference)

**Comment by gregtatcam** [comment]
> @PeterChen13579 this PR currently has merge conflicts which you'll need to resolve (preferably with a merge commit).
> 
> @gregtatcam do you have an opinion on whether it's OK for this PR to be merged before #4626?
> 
> (I have no preference)

I have no preference either.

**Comment by intelliot** [comment]
@PeterChen13579 - please resolve conflicts - preferably by pushing a merge commit (do not force-push).

**Comment by intelliot** [comment]
Suggested commit message:
```
APIv2(gateway_balances, channel_authorize): update errors (#4618)

Since these are breaking changes, they apply only to API version 2.
* Supersedes #4577

gateway_balances
* When `account` does not exist in the ledger, return `actNotFound`
  * (Previously, a normal response was returned)
  * Fix #4290
* When required field(s) are missing, return `invalidParams`
  * (Previously, `invalidHotWallet` was incorrectly returned)
  * Fix #4548

channel_authorize
* When the specified `key_type` is invalid, return `badKeyType`
  * (Previously, `invalidParams` was returned)
  * Fix #4289
```

File: src/ripple/rpc/handlers/GatewayBalances.cpp [hunk 1]
```diff
@@ -78,6 +78,12 @@ doGatewayBalances(RPC::JsonContext& context)
 
     result[jss::account] = toBase58(accountID);
 
+    if (!ledger->exists(keylet::account(accountID)) && context.apiVersion > 1u)
```
Comment (arihantkothari) [inline src/ripple/rpc/handlers/GatewayBalances.cpp:81]: Just noticed here, apiVersion check should be the first check to avoid unnecessary checks.

**Review by arihantkothari — APPROVED** [review]

**Comment by intelliot** [comment]
note: conflict is expected to go away once https://github.com/XRPLF/rippled/pull/4504 is merged

**Comment by intelliot** [comment]
- note: rippled catching up with what Clio already does

**Pull request merged**